models/classes/AdvancedSearch/AdvancedSearchChecker.php

```diff
@@ -27,18 +27,45 @@
 use oat\tao\model\featureFlag\FeatureFlagCheckerInterface;
 use oat\tao\model\search\SearchInterface;
 use oat\tao\model\search\SearchProxy;
+use common_ext_ExtensionsManager;
 
 class AdvancedSearchChecker extends ConfigurableService
 {
     /** @deprecated Use oat\tao\model\featureFlag\FeatureFlagCheckerInterface::FEATURE_FLAG_ADVANCED_SEARCH_DISABLED */
     public const FEATURE_FLAG_ADVANCED_SEARCH_DISABLED = FeatureFlagCheckerInterface::FEATURE_FLAG_ADVANCED_SEARCH_DISABLED;
 
+    public const CONFIG_ADVANCED_SEARCH_DISABLED_SECTIONS = 'TAO_ADVANCED_SEARCH_DISABLED_SECTIONS';
+
     public function isEnabled(): bool
     {
         return !$this->getFeatureFlagChecker()->isEnabled(FeatureFlagCheckerInterface::FEATURE_FLAG_ADVANCED_SEARCH_DISABLED)
             && $this->getSearchService()->supportCustomIndex();
     }
 
+    public function getDisabledSections(): array
+    {
+        $disabledSections = [
+            'results',
+        ];
+
+        if ($this->getExtensionsManager()->isEnabled('taoBooklet')) {
+            $disabledSections[] = 'taoBooklet_main';
+        }
+
+        if (isset($_ENV[self::CONFIG_ADVANCED_SEARCH_DISABLED_SECTIONS])) {
+            $conf = $_ENV[self::CONFIG_ADVANCED_SEARCH_DISABLED_SECTIONS];
+
+            foreach (explode(',', $conf) as $section) {
+                $section = trim($section);
+                if ($section != '' && !in_array($section, $disabledSections)) {
+                    $disabledSections[] = trim($section);
```

Comment:
Already trimmed before

```suggestion
                    $disabledSections[] = $section;
```

Reply:
:man_facepalming: